src/Extensions/ElementalAreasExtension.php

```diff
@@ -223,24 +223,26 @@ protected function onBeforeWrite()
         if (!$this->supportsElemental()) {
             return;
         }
-
-        $elementalAreaRelations = $this->owner->getElementalRelations();
-
+        $owner = $this->getOwner();
+        $elementalAreaRelations = $owner->getElementalRelations();
         $this->ensureElementalAreasExist($elementalAreaRelations);
-
-        $ownerClassName = get_class($this->owner);
+        $ownerClassName = $owner->ClassName;
 
         // Update the OwnerClassName on EA if the class has changed
-        foreach ($elementalAreaRelations as $eaRelation) {
-            $ea = $this->owner->$eaRelation();
-            if ($ea->OwnerClassName !== $ownerClassName) {
-                $ea->OwnerClassName = $ownerClassName;
-                $ea->write();
+        // Do not update if using live versioning mode i.e. publishing,
+        // as it can result in a duplicate ElementalArea being created
+        if (Versioned::get_reading_mode() !== 'Stage.' . Versioned::LIVE) {
```

Comment:
```suggestion
        // Only update if on draft stage, otherwise it can result in a duplicate ElementalArea being created
        if (Versioned::get_stage() === Versioned::DRAFT) {
```

Reply:
Updated